vortex-scalar/src/primitive.rs

```diff
@@ -270,3 +272,47 @@ impl From<usize> for Scalar {
         Scalar::primitive(value as u64, Nullability::NonNullable)
     }
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum NumericOperator {
+    Add,
+    Sub,
+    Mul,
+    Div,
+    // Missing from arrow-rs:
+    // Min,
+    // Max,
+    // Sqrt,
+    // Pow,
+    // In IEEE-754 operations involving NaN are non-commutative wrt NaN payload/quietness
+    // RAdd,
+    // RSub,
+    // RMul,
+    // RDiv,
+}
+
+impl PrimitiveScalar<'_> {
+    pub fn numeric_operator(
+        self,
+        other: PrimitiveScalar<'_>,
+        op: NumericOperator,
+    ) -> VortexResult<Scalar> {
+        if !self.dtype().eq_ignore_nullability(other.dtype()) {
```

Comment:
Why are we ignoring nullability? Not saying we shouldn't be (although maybe we shouldn't be), but if we do this it should be commented.

Reply:
Only because we of the general approach in Vortex that a compute function should never perform type coercion.